app/model/CampaignReferral.scala

```diff
@@ -0,0 +1,79 @@
+package model
+
+import java.time.LocalDate
+
+import com.amazonaws.services.dynamodbv2.document.Item
+import play.api.Logger
+import play.api.libs.json._
+import repositories.CampaignReferralRepository
+
+import scala.util.control.NonFatal
+
+// A referral to any item in a campaign from an on-platform source in a particular period
+case class CampaignReferral(
+  campaignId: String,
+  platform: String,
+  edition: String,
+  path: String,
+  numClicks: Int,
+  containerIndex: Int,
+  containerName: String,
+  cardIndex: Int,
+  cardName: String,
+  firstReferral: LocalDate,
+  lastReferral: LocalDate
+)
+
+object CampaignReferral {
+
+  // dynamodb item -> CampaignReferral
+  implicit val itemReads: Reads[CampaignReferral] = Reads { json =>
+    val platformUrlComponent: Seq[String] = (json \ "platformUrlComponent").as[String].split("\\|").toSeq
+
+    def platform(s: String) = s match {
+      case "NEXT_GEN" => "Web"
+      case _          => "unknown"
```

Comment:
That's all I'm getting at the moment.  I need to tweak my query once I've got the whole thing working end-to-end.